athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java

```diff
@@ -199,10 +200,17 @@ public static Object coerceValueToExpectedType(Object value, Field field,
         }
         if ((fieldType.equals(Types.MinorType.DATEMILLI) || fieldType.equals(Types.MinorType.DATEDAY))
                 && (value instanceof String || value instanceof BigDecimal)) {
-            ZoneId defaultTimeZone = recordMetadata.getDefaultTimeZone();
-            String customerConfiguredFormat = recordMetadata.getDateTimeFormat(field.getName());
+            String dateTimeFormat = recordMetadata.getDateTimeFormat(field.getName());
+            if (value instanceof String && StringUtils.isEmpty(dateTimeFormat)) {
+                logger.info("Date format not in cache for column {}. Trying to infer format...", field.getName());
+                dateTimeFormat = DateTimeFormatterUtil.inferDateTimeFormat((String) value);
+                if (StringUtils.isNotEmpty(dateTimeFormat)) {
+                    logger.info("Adding datetime format {} for column {} to cache", dateTimeFormat, field.getName());
+                    recordMetadata.setDateTimeFormat(field.getName(), dateTimeFormat);
+                }
+            }
```

Comment:
Thanks, keep in mind that this is likely to get refactored a bit when we move this connector to the extractor framework.